ops/model.py

```diff
@@ -2274,7 +2274,7 @@ def _build_destpath(file_path: StrOrPath, source_path: StrOrPath, dest_dir: StrO
         # /src --> /dst/src
         file_path, source_path, dest_dir = Path(file_path), Path(source_path), Path(dest_dir)
         prefix = str(source_path.parent)
-        if os.path.commonprefix([prefix, str(file_path)]) != prefix:
+        if prefix != '.' and os.path.commonprefix([prefix, str(file_path)]) != prefix:
```

Comment:
Reiterating for visibility from the conversation below :)

The `.` prefix is required for relative paths since the parent to source path could be `.`.